test/ex_doc/language/elixir_test.exs

```diff
@@ -439,6 +439,16 @@ defmodule ExDoc.Language.ElixirTest do
     warn(~m"`c:InMemory.unknown/0`")
   end
 
+  test "warning if typespec references filtered module" do
+    ExDoc.Refs.insert([
+      {{:module, AutolinkTest.Keep}, :public},
+      {{:function, AutolinkTest.Filtered}, :public},
+      {{:type, AutolinkTest.Filtered, :type, 0}, :public}
+    ])
+
+    # TODO: testing
```

Comment:
I would like some guidance here, I was able to check the fix with a local project but I'm not sure what would be the best way to test this.

Also, maybe we should create a new test case for this, as it's a bit more complex than the other cases that are based only on autolink